fl4health/server/tabular_feature_alignment_server.py

```diff
@@ -0,0 +1,142 @@
+import random
+from functools import partial
+from logging import DEBUG, INFO, WARNING
+from typing import Callable, Optional, Tuple
+
+from flwr.common.logger import log
+from flwr.common.typing import Config
+from flwr.server.client_manager import ClientManager
+from flwr.server.history import History
+
+from fl4health.checkpointing.checkpointer import TorchCheckpointer
+from fl4health.feature_alignment.constants import (
+    CURRENT_SERVER_ROUND,
+    FEATURE_INFO,
+    FORMAT_SPECIFIED,
+    INPUT_DIMENSION,
+    OUTPUT_DIMENSION,
+)
+from fl4health.feature_alignment.tab_features_info_encoder import TabularFeaturesInfoEncoder
+from fl4health.reporting.fl_wanb import ServerWandBReporter
+from fl4health.server.base_server import FlServer
+from fl4health.server.polling import poll_clients
+from fl4health.strategies.basic_fedavg import BasicFedAvg
+
+
+class TabularFeatureAlignmentServer(FlServer):
+    """
+    This server is used when the clients all have tabular data that needs to be
+    aligned.
+
+    tab_features_source_of_truth: the information that is required for aligning client features.
+    If it is not specified, then the server will randomly poll a client and gather
+    this information from its data source.
+    """
+
+    def __init__(
+        self,
+        client_manager: ClientManager,
+        config: Config,
+        initialize_parameters: Callable,
+        strategy: BasicFedAvg,
+        wandb_reporter: Optional[ServerWandBReporter] = None,
+        checkpointer: Optional[TorchCheckpointer] = None,
+        tabular_features_source_of_truth: Optional[TabularFeaturesInfoEncoder] = None,
+    ) -> None:
+        assert isinstance(strategy, BasicFedAvg)
```

Comment:
Since we're explicitly typing this argument in the `__init__` signature, do we still need this assert?